bigquery_etl/alchemer/survey.py

```diff
@@ -12,35 +12,64 @@
 from requests import HTTPError
 
 
-def utc_date_to_eastern_string(date_string):
+def utc_date_to_eastern_string(date_string: str) -> str:
     """Normalize ISO date to UTC midnight."""
     naive_dt = dt.datetime.strptime(date_string, "%Y-%m-%d")
     as_utc = pytz.utc.localize(naive_dt)
     as_eastern = as_utc.astimezone(pytz.timezone("US/Eastern"))
     return dt.datetime.strftime(as_eastern, "%Y-%m-%d+%H:%M:%S")
 
 
-def date_plus_one(date_string):
+def date_plus_one(date_string: str) -> str:
     """Get the day after the current date."""
     return dt.datetime.strftime(
         dt.date.fromisoformat(date_string) + dt.timedelta(days=1), "%Y-%m-%d"
     )
 
 
-def format_responses(s, date):
-    """Return a nested field of responses for each user."""
-    # `survey_data` is a dict with question ID as the key and question/response
-    # details as the value e.g. survey_data: {'25': {'id': 25, 'type': 'RADIO',
-    # 'question': 'I trust Firefox to help me with my online privacy',
-    # 'section_id': 2, 'answer': 'Agree', 'answer_id': 10066, 'shown': True}}
-    # See https://apihelp.alchemer.com/help/surveyresponse-returned-fields-v5#getobject
+def make_date_iso_compliant(date_string: str) -> str:
+    """Convert date to ISO-compliant format. E.g. '2018-09-27 10:42:16 EDT' becomes 2018-09-27 10:42:16 -04:00'."""
+    tz_mapping = {"EDT": "-04:00", "EST": "-05:00"}
+    tz_match = re.search("E[DS]T", date_string)
```

Comment:
BigQuery is going to consider the timestamp strings generated by this code to be invalid due to the space between the time part and the offset.
```suggestion
    """Convert date to ISO-compliant format. E.g. '2018-09-27 10:42:16 EDT' becomes 2018-09-27 10:42:16 -04:00'."""
    tz_mapping = {" EDT": "-04:00", " EST": "-05:00"}
    tz_match = re.search(" E[DS]T$", date_string)
```

Reply:
Based on manually testing a load job it looks like you're right!

That's surprising because [some relevant BigQuery docs](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#utc_offset) say
> You can specify the offset from UTC using the following format: `{+|-}H[H][:M[M]]`
> When using this format, no space is allowed between the time zone and the rest of the timestamp.

And running a query like `SELECT TIMESTAMP('2018-09-27 10:42:16 -04:00')` fails with an `Invalid timestamp` error.

I guess BigQuery is currently less strict parsing timestamps in load jobs than it is in normal query jobs, though that might be considered a bug in some sense, so I'd still suggest adjusting this code to produce timestamp values that match what BigQuery's docs say is the expected timestamp format.